### ISSUE TITLE
Add the possibility of a visible grid in the net editor

### ISSUE THREAD

**Issue by MartinBeseda** [spirali/kaira#12]
Now there's only the possibility on the invisible grid, which is not so helpful, because you can't think the net through so well, when you can't see it. It'd be great to have a visible grid in the editor on which you can attach elements of your net.

**Comment by msurkovsky** [comment]
Well I don't think it is necessary to show a grid. In my opinion, such a functionality usually makes diagrams less clear. Nevertheless, I see a problem with too fine grid, that's uncomfortable for me. So, maybe it would be enough to provide a better values for grid sizes.

**Comment by MartinBeseda** [comment]
I didn't mean a permanent grid, which is a part of the net. I thought about the grid which can be easily shown/hidden and which is not "rendered" to the simulation etc.

Better grid sizes, or the possibility of changing this size, would be very helpful, too. I completely agree with this one.

**Comment by msurkovsky** [comment]
I'm aware that many tools provide showing/hiding a grid, but in my opinion it's usually more annoying than useful. But, if it is implemented (and showed only on demand) I won't have problem with this. I would more appreciate the possibility to set the grid size.

**Comment by spirali** [comment]
I agree, that there should be some more sophisticated grid options, but I am thinking about behaviour implemented in CPN Tools and their "smart snapping".

But for me, at least a visible grid looks good to me.

**Comment by msurkovsky** [comment]
The "smart snapping" can be also a way, but from my point of view there is a lot of space for "mistakes" (= sometimes it works, sometimes not) and the result is that is annoying behavior. In my opinion a possibility to set grid size is good enough. Nevertheless, when we are talking about making editting more comfortable, I would be for a better mechanism of switching between tools (new place, transition, etc.). Somehow improve the editing so that the creation of new programs will be faster. At least some shortcuts or context menus.
